src-java/stats-topology/stats-storm-topology/src/main/java/org/openkilda/wfm/topology/stats/service/KildaEntryCacheService.java

```diff
@@ -257,8 +345,29 @@ private void refreshCommonFlowsCache() {
                 });
     }
 
+    private void refreshHaFlowsCache() {
+        CacheAddUpdateHandler cacheHandler =
+                new CacheAddUpdateHandler(cookieToFlow, switchAndMeterToFlow, switchAndGroupToFlow);
+        haFlowRepository.findAll()
+                .forEach(haFlow ->
```

Comment:
usages of multi-layered lambda were discouraged in other reviews. Please, consider to extract lambdas to private methods and use method references here

Reply:
thanks, I extracted methods, but I can not see any chances to use method references here, If you have any suggestions that I do not see, please let me know.

Anyway, imho,  we might not need here any breaking down a multi layer lambda into private methods, I do not see any readability improvements, to be honest. 

This is updated version:
```
  private void refreshHaFlowsCache() {
        CacheAddUpdateHandler cacheHandler =
                new CacheAddUpdateHandler(cookieToFlow, switchAndMeterToFlow, switchAndGroupToFlow);
        haFlowRepository.findAll().forEach(haFlow -> processHaFlow(haFlow, cacheHandler));
    }

    private void processHaFlow(HaFlow haFlow, CacheAddUpdateHandler cacheHandler) {
        haFlow.getPaths().forEach(haPath -> processHaPath(haPath, cacheHandler));
    }

    private void processHaPath(HaFlowPath haPath, CacheAddUpdateHandler cacheHandler) {
        haPath.getSubPaths().forEach(subPath -> updateCacheHa(
                cacheHandler, subPath.getHaFlowId(),
                FlowPathMapper.INSTANCE.mapToPathNodes(subPath.getHaFlowPath().getHaFlow(), subPath),
                subPath.getCookie(),
                subPath.getMeterId(),
                subPath.getCookie().getDirection() == FlowPathDirection.FORWARD
                        ? subPath.getHaFlowPath().getYPointGroupId() : null,
                subPath.getCookie().getDirection() == FlowPathDirection.REVERSE
                        ? subPath.getHaFlowPath().getYPointMeterId() : null,
                subPath.getHaSubFlowId(),
                subPath.getHaFlowPath().getYPointSwitchId()));
    }
``` 

anyway, let it be so.

Reply:
Well, what I mean by "consider" is to think if there is a benefit in this case. Usually it is easier to understand and read when it is split into smaller private methods. If no, then no is a good answer as well. I am not insisting.
About the method references, you probably can't do that because of you pass `cacheHandler` in the first and in the second method in the chain, while you don't actually need it there; maybe you can move that handler closer to the place where it is used: into `processHaPath`. In that case your method would have signature on Consumer which is what you pass into forEach.
But, if you believe it is overcomplicating things, then just let it be as is.